### PR TITLE
docs: update wording

### PR DESCRIPTION
**Pull request by craigbilner**
Fixed some typos and made some perhaps more controversial wording changes.

I have these observations too, what would be the best way to raise these?

* A bit odd that the JS/Dart comparison is in the React Native section not the web dev section or a completely separate one?
* What is the equivalent of the React Native Hello World app in Flutter? - The Flutter one appears to "render" the app but the "render" logic isn't in the RN example?
* `CustomCard({@required this.index, @required this.onPress});` essentially has `PropTypes` but the RN example doesn't have them?

### CODE REVIEW

**Comment by craigbilner** [comment]
I signed it!

File: flutter-for-android.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ every new frame, Flutter's framework creates a new tree of widget instances. In
 comparison, an Android view is drawn once and does not redraw until invalidate
 is called.
 
-Unlike Android’s view hierarchy system where the framework mutate views,
+Unlike Android’s view hierarchy system where the framework mutates views,
```
Comment (rock3r) [inline flutter-for-android.md]: Ops, my bad! Nice catch 👍

File: cookbook/navigation/returning-data.md [hunk 2]
```diff
@@ -123,7 +123,7 @@ class SelectionScreen extends StatelessWidget {
 
 Now, we'll want to update the `onPressed` callback for both of our buttons! In
 order to return data to the first screen, we'll need to use the 
-[`Navitator.pop`](https://docs.flutter.io/flutter/widgets/Navigator/pop.html)
+[`Navigator.pop`](https://docs.flutter.io/flutter/widgets/Navigator/pop.html)
```
[the remark this answers is not in the document]
Comment (brianegan) [reply cookbook/navigation/returning-data.md]: Ouch.

File: flutter-for-react-native.md [hunk 3]
```diff
@@ -430,7 +430,7 @@ new CustomCard(
     ...
 {% endprettify %}
 
-The constructor for the `CustomCard` class shows a Dart feature. The curly braces inside the constructor indicates that the parameters are optional when initialising. In order to make these fields required, we have two options- first one is to remove the curly braces from the constructor and the second is to add the `@required` to the constructor. The latter approach enables the developer to provide parameter names when using the `CustomCard` class in the app code as show in the (Usage section) example above. 
+The constructor for the `CustomCard` class shows a Dart feature. The curly braces inside the constructor indicates that the parameters are optional when initialising. In order to make these fields required, we have two options- first one is to remove the curly braces from the constructor and the second is to add the `@required` to the constructor. The latter approach enables the developer to provide parameter names when using the `CustomCard` class in the app code as shown in the (Usage section) example above.
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: I see no difference in these paragraphs and it's in the same location, but I do see some grammar errors. Since the errors were there already, it's fine.

File: cookbook/persistance/key-value.md [hunk 4]
```diff
@@ -114,9 +114,8 @@ class _MyHomePageState extends State<MyHomePage> {
   //Incrementing counter after click
   _incrementCounter() async {
     SharedPreferences prefs = await SharedPreferences.getInstance();
-    _counter = (prefs.getInt('counter') ?? 0) + 1;
     setState(() {
-      _counter;
+      _counter = (prefs.getInt('counter') ?? 0) + 1;
```
[the remark this answers is not in the document]
Comment (brianegan) [reply cookbook/persistance/key-value.md]: Looks good to me. Several more updates coming to this recipe as well: #920 !

File: flutter-for-react-native.md [hunk 5]
```diff
@@ -634,7 +634,7 @@ render() {
 }
 {% endprettify %}
 
-In Flutter, we use [`CustomPaint`](https://docs.flutter.io/flutter/widgets/CustomPaint-class.html) widget to draw during the paint phase. We implement abstract class [`CustomPainter`](https://docs.flutter.io/flutter/rendering/CustomPainter-class.html) and pass it to `painter` property in `CustomPaint` widget.
+In Flutter, we use the [`CustomPaint`](https://docs.flutter.io/flutter/widgets/CustomPaint-class.html) widget to draw during the paint phase. We implement abstract class [`CustomPainter`](https://docs.flutter.io/flutter/rendering/CustomPainter-class.html) and pass it to `painter` property in `CustomPaint` widget.
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: Good catch.

File: flutter-for-react-native.md [hunk 6]
```diff
@@ -1121,7 +1121,7 @@ class MyStatelessWidget extends StatelessWidget {
 
 ### Props
 
-In React Native, most components can be customized when they are created with different parameters. These creation parameters are called props. These parameters can be used in child component using `this.props`.
+In React Native, most components can be customized when they are created with different parameters. These creation parameters are called props. These parameters can be used in a child component using `this.props`.
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: Good catch.

File: cookbook/persistance/key-value.md [hunk 7]
```diff
@@ -166,4 +165,4 @@ const MethodChannel('plugins.flutter.io/shared_preferences')
   });
 ```
 
-The code above should be placed inside test file under `test` folder.
\ No newline at end of file
+The code above should be placed inside a test file under the `test` folder.
```
Comment (brianegan) [inline cookbook/persistance/key-value.md]: Can we revert this change? I've fixed this and done some rearranging in #920

File: flutter-for-react-native.md (hunk 3 above)
Comment (rock3r) [reply flutter-for-react-native.md]: There is a `show` -> `shown` change towards the end of the last sentence

File: tutorials/layout/index.md [hunk 8]
```diff
@@ -1482,7 +1482,7 @@ class _MyHomePageState extends State<MyHomePage> {
 ### Card
 
 A Card, from the Material Components library, contains related nuggets of information
-and can be composed from most any widget, but is often used with ListTile.
+and can be composed from almost any widget, but is often used with ListTile.
```
Comment (Sfshaza) [inline tutorials/layout/index.md]: That's fine.

File: tutorials/interactive/index.md [hunk 9]
```diff
@@ -41,7 +41,7 @@ that manages two stateless widgets.
 
 ## Getting ready
 
-If you've already build the layout in
+If you've already built the layout in
```
Comment (Sfshaza) [inline tutorials/interactive/index.md]: Good catch.

File: flutter-for-react-native.md (hunk 3 above)
Comment (Sfshaza) [reply flutter-for-react-native.md]: Good catch!  (as show => as shown)

File: web-analogs.md [hunk 10]
```diff
@@ -665,7 +664,8 @@ var container = new Container( // grey box
 
 ### Making Circles and Ellipses
 Making a circle in CSS requires a workaround of applying a border-radius of
-50% to all four sides of a rectangle.
+50% to all four sides of a rectangle (although there are [basic-shapes](https://developer.mozilla.org/en-US/docs/Web/CSS/basic-shape)
```
Comment (Sfshaza) [inline web-analogs.md]: That's fine.

File: web-analogs.md [hunk 11]
```diff
@@ -157,9 +157,8 @@ var container = new Container( // grey box
 ### Setting Container Width
 
 To specify the width of a [Container](https://docs.flutter.io/flutter/widgets/Container-class.html)
-widget you set its ```width``` property. This is a fixed width, unlike the
-CSS max-width property which adjusts container width up to a maximum value. To
-mimic that effect in Flutter, use the ```constraints``` property of the Container.
+widget you set its ```width``` property just like in CSS. To
+mimic CSS's ```max-width``` in Flutter, use the ```constraints``` property of the Container.
```
Comment (Sfshaza) [inline web-analogs.md]: Oy. Suddenly this page switches from one backtick for code to 3. This is incorrect. It should be one. But you don't have to fix it.

Comment (Sfshaza) [reply web-analogs.md]: "just like in CSS" => as you would in CSS.
delete "in Flutter"
Please replace all ``` with `.

File: flutter-for-react-native.md [hunk 12]
```diff
@@ -1814,7 +1815,7 @@ class FadeInView extends React.Component {
...
+In the equivalent Flutter example, we use an [`AnimationController`](https://docs.flutter.io/flutter/animation/AnimationController-class.html) object `controller` and specify the duration inside it. The animation is defined as a [`Tween`](https://docs.flutter.io/flutter/animation/Tween-class.html) which basically describes some interpolation between a beginning and ending value. We are using a [`FadeTransition`](https://docs.flutter.io/flutter/widgets/FadeTransition-class.html) widget where we have `opacity` property which is mapped to the `animation` object. The sole job of a Tween is to define a mapping from an input range to an output range. Then we start the animation using `controller.forward()`. We can perform other operations too using the controller. In the example, We are using the [`FlutterLogo`](https://docs.flutter.io/flutter/material/FlutterLogo-class.html) inside the `FadeTransition` widget.
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: Can you show the change here? It highlights the entire paragraph.

File: flutter-for-react-native.md [hunk 13]
```diff
@@ -1707,7 +1707,8 @@ void _submit() {
 
 ## Platform-specific code
 
-When building a cross-platform app, you'll want to re-use as much code as possible across platforms. Scenarios may arise where it makes sense for the code to be different. In React Native, separate implementation can be given at almost anywhere in the app code by recognizing the platform the app will run on. To get the target platform, you need to use this component :
+When building a cross-platform app, you'll want to re-use as much code as possible across platforms. Scenarios may arise where it makes sense for the code to be different. In React Native, separate implementation can be given at almost anywhere in the app code by recognizing the platform the app will run on. To get the target platform,
+you can either use platform file extensions or the `Platform` module as below:
```
Comment (Sfshaza) [inline flutter-for-react-native.md]: In React Native, separate... => In React Native, a separate...
either use platform file extensions or => either use platform file extensions, or

Another problem I'm noticing. Lines of text should never be more than 80 chars long. You don't have to fix pre-existing lines, but please make all new lines 80 chars.

**Comment by Sfshaza** [comment]
Thanks, @craigbilner, for tidying up our docs! (And, I see,  they need more tidying.  ;) )

I've called out a couple minor fixes.

File: web-analogs.md [hunk 14]
```diff
@@ -801,7 +801,7 @@ property.
 
 In Flutter, you transform the contents of a Text widget using the methods and
 operators of the [String class](https://docs.flutter.io/flutter/dart-core/String-class.html)
-in the dart:core library.
+in the dart:core library as you would with JavaScript instead.
```
[the remark this answers is not in the document]
Comment (craigbilner) [reply web-analogs.md]: I found it quite an odd comparison? If there was a `TextTransformWidget` then a CSS comparison would make sense, but essentially this is saying, don't use "styling" but change the underlying data a la JS?

File: flutter-for-react-native.md (hunk 12 above)
Comment (craigbilner) [reply flutter-for-react-native.md]: We animation => The animation

Although perhaps `we have opacity property` is a little odd too

File: flutter-for-react-native.md (hunk 13 above)
Comment (craigbilner) [reply flutter-for-react-native.md]: Would it be worth adding prettier to the repo to automate this away? Happy to raise a PR.

**Comment by craigbilner** [comment]
np @Sfshaza, think I've addressed everything

**Comment by Hixie** [comment]
@Sfshaza Can you please take another look? Thanks!

**Comment by Sfshaza** [comment]
Hey, @craigbilner!

Some of these issues have been addressed, some are being addressed in another PR, some have not yet been addressed. I've created a new [PR](https://github.com/flutter/website/pull/1052) to deal with the issues that I can deal with now.

There's an outstanding PR for the React Native page. Thanks for submitting these errors! (Though, in future, it's much easier for us to address them if you just file the PR on one file - or a couple related files. Less balls in the air.

Thanks!
